editor/store/selectors.js

```diff
@@ -563,6 +563,45 @@ export const getBlocks = createSelector(
 	]
 );
 
+const unfoldBlockIds = ( state, block ) => {
+	const clientIds = [];
+	if ( block.name === 'core/block' ) {
+		clientIds.push( get( getSharedBlock( state, block.attributes.ref ), [ 'clientId' ], null ) );
+	}
+	if ( block.innerBlocks.length > 0 ) {
+		block.innerBlocks.map( ( innerBlock ) => {
+			clientIds.push( innerBlock.clientId );
+			clientIds.push( ...unfoldBlockIds( state, innerBlock ) );
+		} );
+	}
+	return clientIds.filter( ( id ) => !! id );
+};
+
+/**
+ * Returns an array containing the ids of the top-level blocks,
+ * all blocks referenced by the existing top-level shared blocks,
+ * and all the inner blocks of existing top-level nested blocks.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {Array} All ids of top-level, referenced, and inner blocks.
+ */
+export const getBlockIdsUnfolded = createSelector(
```

Comment:
Maybe we can call this select getAllBlockClientIdsReferenced and specify it returns the array of all block uids referenced/used in the document even the ones inside shared blocks.

Reply:
Renamed to `getClientIdsUnfolded` which is in line with the other selector names (get + thing + modifier).